### PR TITLE
Add a post_integrate_hook API

### PR DESCRIPTION
**Pull request by lucasmpaim**
Create a post_integrate API, with this we are able to add a scripts to execute after the project are generated, like:

```ruby
post_integrate do |installer|
    puts "Called"
    bricklayer.configure_project
end
```

obs: `bricklayer` is my own package to link and compile project subproject's, for execute this I need the `.xcworkspace` file already created

Ref. [#7432](https://github.com/CocoaPods/CocoaPods/issues/7432#issuecomment-561745032)

DSL PR: [#606](https://github.com/CocoaPods/Core/pull/606)

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Add a post_integrate_hook API  
  [lucasmpaim](https://github.com/lucasmpaim)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by amorde** [comment]
@lucasmpaim can you run `bundle update cocoapods-core` and commit `Gemfile.lock`? I'm also happy to do this step for you if you'd like

File: Gemfile.lock [hunk 1]
```diff
@@ -7,11 +7,12 @@ GIT
 
 GIT
   remote: https://github.com/CocoaPods/Core.git
-  revision: b84eb6ab23a3e65a0ab79487fd3fe44c590cb29a
+  revision: 04d111d9deb3e09563933ae547c7c6f1a104b6a4
   branch: master
   specs:
-    cocoapods-core (1.9.0.beta.3)
+    cocoapods-core (1.9.0)
```
Comment (dnkoutso) [inline Gemfile.lock]: this is weird as 1-9-stable is already merged into master https://github.com/CocoaPods/CocoaPods/blob/master/Gemfile.lock#L13

Comment (dnkoutso) [reply Gemfile.lock]: is your branch properly rebased against latest master?

Comment (dnkoutso) [reply Gemfile.lock]: still probably need to bump the SHA of cocoapods-core to ensure your other PR is available to you with the DSL.

Comment (lucasmpaim) [reply Gemfile.lock]: Sorry @dnkoutso, I have made the merge again, this branch it's outdated

**Comment by lucasmpaim** [comment]
Create the PR that can fix the travis including the new output line on tests [#272](https://github.com/CocoaPods/cocoapods-integration-specs/pull/272)

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
-* None.  
+* Add a post_integrate_hook API  
+  [lucasmpaim](https://github.com/lucasmpaim)
+  [#7432](https://github.com/CocoaPods/CocoaPods/issues/7432)
+
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit of nits remove this extra line

Comment (lucasmpaim) [reply CHANGELOG.md]: fixed :)

**Comment by dnkoutso** [comment]
@lucasmpaim merged the integration specs PR, please update this PR with the submodule reference so it can go green.

**Comment by lucasmpaim** [comment]
I found two files that not alert before on my local tests, description are in [#273](https://github.com/CocoaPods/cocoapods-integration-specs/pull/273)

**Comment by lucasmpaim** [comment]
@dnkoutso now all papelines on travis are passing, execept one:

Screen Shot 2020-02-28 at 21 43 09

it appears that have something deprecated being called, I'm not sure what I need to do this become true now, this is working on master branch? that's is occurring in a file that I don't have add any changes

**Comment by dnkoutso** [comment]
I rekicked your build lets see..

**Comment by dnkoutso** [comment]
@lucasmpaim can you also squash into a single commit?

**Comment by lucasmpaim** [comment]
@dnkoutso I have made the squash :)

**Comment by lucasmpaim** [comment]
Finally it's green!!!  😁

**Pull request merged**